src/Language/PureScript/Ide/Types.hs

```diff
@@ -66,37 +65,71 @@ data ExternDecl
     | Export ModuleIdent -- The exported Modules name
     deriving (Show,Eq,Ord)
 
-type Module = (ModuleIdent, [ExternDecl])
+data IdeDeclaration
+  = IdeValue Ident P.Type
+  | IdeType (P.ProperName 'P.TypeName) P.Kind
+  | IdeTypeSynonym (P.ProperName 'P.TypeName) P.Type
+  | IdeDataConstructor Ident (P.ProperName 'P.TypeName) P.Type
+  | IdeTypeClass (P.ProperName 'P.ClassName)
+  | IdeValueOperator (P.OpName 'P.ValueOpName) Ident P.Precedence P.Associativity
+  | IdeTypeOperator (P.OpName 'P.TypeOpName) Ident P.Precedence P.Associativity
+  deriving (Show, Eq, Ord)
+
+type Module = (P.ModuleName, [IdeDeclaration])
+type ModuleOld = (Text, [ExternDecl])
```

Comment:
Could you please add some comments to these, and this one in particular? I'm confused why there are two module types.